test/jobs/process_opt_out_test.exs

```diff
@@ -48,16 +52,18 @@ defmodule Companion.Jobs.ProcessOptOutTests do
           {"user-agent", "nurseconnect-companion"},
           {"content-type", "application/json"}
         ],
-        body: @openhim_expected_request
+        body: body
       } ->
-        %Tesla.Env{status: 202, body: "Accepted"}
+        json(%{})
     end)
 
-    :ok
+    response
   end
 
   test "sends opt out to OpenHIM" do
     {:ok, optout} = create_opt_out(%{contact_id: "a49fddb7-cde0-4d3a-aa24-33ecc826f0d2"})
+    reg_request = Map.put(@openhim_expected_request, :eid, optout.id)
```

Comment:
I'm not sure how this test is passing, since `eid` should now be a UUID instead of just the integer, so I think we should investigate how this is still passing.

Reply:
I think its because I keep getting this warning in the test ```warning: variable "body" is unused
  test/jobs/process_opt_out_test.exs:36``` where all the fields are in the body, but I dont know how to fix it

Reply:
I think that's in the `mock_request` function. The way the mock function is called the static analysis doesn't pick it up that it's being used. You should be able to fix it by returning `response, body` on line 60, instead of just `response`. 

But I'm not sure that that tells us why the test is passing when it should be failing.